src/nupic/algorithms/Connections.hpp

```diff
@@ -436,6 +436,17 @@ class Connections : public Serializable
    */
   void bumpSegment(const Segment segment, const Permanence delta);
 
+  /**
+   * Destroy the synapses with the lowest permanence values.  This method is
+   * useful for making room for more synapses on a segment which is already
+   * full.
+   *
+   * @param segment - Index of segment in Connections, to be modified.
+   * @param nDestroy - Must be greater than or equal to zero!
+   * @param excludeCells - Presynaptic cells which will NOT have any synapses destroyed.
+   */
+  void destroyMinPermanenceSynapses(const Segment segment, Int nDestroy,
```

Comment:
this method should now have its unit-test, given its now public in Connections.

Reply:
..probably the test could be copied from TM tests, if there were one

Reply:
The TM has unit tests which call this method, albeit indirectly via a method in the TM.  I think that the testing coverage is good enough so I'm going to go ahead and merge this.

Thanks for reviewing this!